File: router-annotations/src/main/java/com/oayilix/lodestar/annotations/Destination.java
package com.oayilix.lodestar.annotations;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

@Target({ElementType.TYPE}) // 元注解，说明当前注解可以修饰的元素，此处标识可以用于标记在类上面
@Retention(RetentionPolicy.CLASS) // 元注解，说明当前注解的生命周期。也就是可以保留的时间。保留到编译为 class 文件。
public @interface Destination {

    /**
     * 当前页面定义的 url，不能为空
     * @return 页面定义的 url
     */
    String url();

    /**
     * 定义当前页面的描述
     * @return 页面描述内容
     */
    String description() default "no description";
}

File: router-annotations/src/main/kotlin/com/oayilix/lodestar/annotations/Destination.kt
package com.oayilix.lodestar.annotations

/**
 * 标记页面路由目标。
 * 编译期注解处理器会扫描所有标注了此注解的类，生成路由映射表。
 *
 * @property url 页面定义的 url，不能为空
 * @property description 定义当前页面的描述
 */
@Target(AnnotationTarget.CLASS)
@Retention(AnnotationRetention.BINARY)
annotation class Destination(
    val url: String,
    val description: String = "no description"
)
